url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/graph/EdgeView.java

```diff
@@ -0,0 +1,1787 @@
+				{
+					if (EdgeView.LEGACY_DISCONNECTABLE)
+					{
+						currentPoint = edge.getPoint(index);
+						currentIndex = index;
+						source = index == 0;
+						target = index == r.length - 1;
+						break;
+					}
+					else
+					{
+						if ((index > 0 && index < r.length - 1)
+								|| GraphConstants.isDisconnectable(edge
+										.getAllAttributes()))
+						{
+							currentPoint = edge.getPoint(index);
+							currentIndex = index;
+							source = index == 0;
+							target = index == r.length - 1;
+							break;
+						}
+						else
+						{
+							event.consume();
+						}
+					}
+				}
+			}
+			
+			// Detect hit on label
+			if (!isEditing() && graph.isMoveable()
+					&& GraphConstants.isMoveable(edge.getAllAttributes())
+					&& loc != null && loc.contains(x, y)
+					&& !isAddPointEvent(event) && !isRemovePointEvent(event)
+					&& graph.getEdgeLabelsMovable()) {
+				initialLabelLocation = (Point2D) edge.getLabelPosition()
+						.clone();
+				label = true;
+			}
+			// Detect hit on extra labels
+			else if (extraLabelLocations != null && !isEditing()
+					&& graph.isMoveable() && graph.getEdgeLabelsMovable()
+					&& GraphConstants.isMoveable(edge.getAllAttributes())) {
+				for (int i = 0; i < extraLabelLocations.length; i++) {
+					if (extraLabelLocations[i] != null
+							&& extraLabelLocations[i].contains(x, y)) {
+						currentLabel = i;
+						initialLabelLocation = (Point2D) edge
+								.getExtraLabelPosition(currentLabel).clone();
+						if (isRemovePointEvent(event)) {
+							edge.removeExtraLabel(i);
+							edgeModified = true;
+							mouseReleased(event);
+						}
+						break;
+					}
+				}
+			}
+			// Remove Point
+			if (isRemovePointEvent(event)
+					&& currentPoint != null
+					&& !source
+					&& !target
+					&& bendable
+					&& (edge.getSource() == null || currentIndex > 0)
+					&& (edge.getTarget() == null || currentIndex < edge
+							.getPointCount() - 1)) {
+				edge.removePoint(index);
+				edgeModified = true;
+				mouseReleased(event);
+				// Add Point
+			} else if (isAddPointEvent(event) && !isEditing() && bendable) {
+				int s = graph.getHandleSize();
+				Rectangle2D rect = graph.fromScreen(new Rectangle(x - s, y - s,
+						2 * s, 2 * s));
+				if (edge.intersects(graph, rect)) {
+					Point2D point = graph.fromScreen(graph.snap(new Point(event
+							.getPoint())));
+					double min = Double.MAX_VALUE, dist = 0;
+					for (int i = 0; i < edge.getPointCount() - 1; i++) {
+						Point2D p = edge.getPoint(i);
+						Point2D p1 = edge.getPoint(i + 1);
+						dist = new Line2D.Double(p, p1).ptSegDistSq(point);
+						if (dist < min) {
+							min = dist;
+							index = i + 1;
+						}
+					}
+					edge.addPoint(index, point);
+					edgeModified = true;
+					currentPoint = point;
+					reloadPoints(edge);
+					paint(graph.getGraphics());
+				}
+			}
+			if (isEditing())
+				event.consume();
+		}
+
+		public void mouseDragged(MouseEvent event) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 50. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.